arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Iterable.kt

```diff
@@ -893,19 +893,13 @@ public fun <A, K, V> Iterable<A>.crosswalkMap(f: (A) -> Map<K, V>): Map<K, List<
       ior.fold(
         { listOf(it) },
         ::identity,
-        { l, r -> listOf(l) + r }
+        { l, r -> r + l }
       )
     }
   }
 
 public fun <A, B> Iterable<A>.crosswalkNull(f: (A) -> B?): List<B>? =
-  fold<A, List<B>?>(emptyList()) { bs, a ->
-    Ior.fromNullables(f(a), bs)?.fold(
-      { listOf(it) },
-      ::identity,
-      { l, r -> listOf(l) + r }
-    )
-  }
+  mapNotNull(f).takeIf { it.isNotEmpty() || it.isEmpty() }
```

Comment:
I think you mean:
```suggestion
  mapNotNull(f).takeIf { it.isNotEmpty() || this.isEmpty() }
```
because otherwise you've just written `mapNotNull(f).takeIf { true }`

Reply:
`Iterable<T>` does not have `isEmpty` method and it is `Collection<T>` method.
So, I will make this back to `!this.any()`

Reply:
I've make it back to `!this.any()` at commit https://github.com/arrow-kt/arrow/commit/7eb453ec2f56519a736cfea54c3b2ff519baa60a